llvm/lib/ObjCopy/ELF/ELFObjcopy.cpp

```diff
@@ -623,6 +623,58 @@ handleUserSection(const NewSectionInfo &NewSection,
   return F(NewSection.SectionName, Data);
 }
 
+static Error verifyNoteSection(StringRef Name, endianness Endianness,
+                               ArrayRef<uint8_t> Data) {
+  // An ELF note has the following structure:
```

Comment:
Random thought: I wonder if there's a way to leverage existing code in the `Object` library or llvm-readobj to verify the note, rather than duplicating things? (There might not be a sensible way of doing this, I haven't looked).

Reply:
It seems possible to hook in ` llvm::objcopy::elf::Object::addSection `, but then it would also be a good opportunity to move the conversion to SHT_NOTE based on the section name there too... Let's investigate that in a follow-up PR

Reply:
Note: one note section may contain multiple notes.
The practical example is `.note.gnu.property`.

There is some alignment complexity. See https://reviews.llvm.org/D150022

Reply:
I always get confused about .note and alignment behaviours. Are you suggesting @MaskRay that some action needs taking here?

Reply:
Some note sections (`.note.gnu.property`) have 8-byte alignment. Perhaps add a test that they are supported. @serge-sans-paille